src/spi.rs

```diff
@@ -34,8 +34,12 @@ where
     }
 
     /// Initializes the BME280
-    pub fn init<D: DelayUs>(&mut self, delay: &mut D) -> Result<(), Error<SPIError<SPIE, PinE>>> {
-        self.common.init(delay)
+    pub fn init<D: DelayUs>(
+        &mut self,
+        delay: &mut D,
+        config: Configuration,
+    ) -> Result<(), Error<SPIError<SPIE, PinE>>> {
+        self.common.init(delay, config)
```

Comment:
And same here similar to the i2c case?
Just so we dont mess around with public API too much?